src/resource/archive/O2rArchive.cpp

```diff
@@ -89,7 +90,7 @@ bool O2rArchive::Open() {
 }
 
 bool O2rArchive::Close() {
-    if (zip_close(mZipArchive) == -1) {
+    if (mZipArchive != nullptr && zip_close(mZipArchive) == -1) {
```

Comment:
i was thinking something like

```suggestion
    if (mZipArchive == nullptr) {
        SPDLOG_ERROR("Cannot close zip file. Zip file not loaded. \"{}\"", GetPath());
        return false;
     }
     
     if (zip_close(mZipArchive) == -1) {
```

Reply:
oh ok